airbyte-integrations/connectors/source-shopify/source_shopify/http_request.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+from typing import Optional, Union
+
+import requests
+from airbyte_cdk.sources.streams.http.error_handlers import ErrorHandler, ErrorResolution, ResponseAction
+from airbyte_protocol.models import FailureType
+from requests import exceptions
+
+RESPONSE_CONSUMPTION_EXCEPTIONS = (
+    exceptions.ChunkedEncodingError,
+    exceptions.JSONDecodeError,
+)
+
+TRANSIENT_EXCEPTIONS = (
+    exceptions.ConnectTimeout,
+    exceptions.ConnectionError,
+    exceptions.HTTPError,
+    exceptions.ReadTimeout,
+    # This error was added as part of the migration from REST to bulk (https://github.com/airbytehq/airbyte/commit/f5094041bebb80cd6602a98829c19a7515276ed3) but it is unclear in which case it occurs and why it is transient
```

Comment:
Can we document why this is transient? @bazarnov

Reply:
During the migration to `GrapghQL BULK` we've noticed the OSS-related, `ConnectionError`, which was the type of `SSLError`, raised when `ConnectionError` apeared along side with the default socket `ConnectionError`, since it's a non-actionable, but retryable error, we decided to retry it, thus - it's transient.